routes/api/v1/favorites.js

```diff
@@ -0,0 +1,58 @@
+var express = require('express');
+var router = express.Router();
+var fetch = require('node-fetch');
+var fetch = require('node-fetch');
+
+var pry = require('pryjs');
+var User = require('../../../models').User;
+var Location = require('../../../models').Location;
+
+router.post("/", function(req, res) {
+  let locationUrl = `https://maps.googleapis.com/maps/api/geocode/json?address=${req.body.location}&key=${process.env.GEOCODING_API_KEY}`;
+
+  User.findOne({
+    where: {
+      api_key: req.body.api_key
+    }
+  })
+    .then ( user => {
+      if(!user){
+        res.sendStatus(401);
```

Comment:
I would check to make sure if you send a bad key you get the 401 status, as far as im aware .findOne functions are promises, so if it does not find a user it might not hit this .then and insted hit the catch on line 56, which will just return the standard error for the .findOne promise. Just a hunch however so if you did test it already then ignore this!

Reply:
Good catch! I tried it and it is returning that 401 status. I looked into the sequelize docs and it looks like it'll always return either an instance of the user or null, so it should always hit that .then (I think!). Thanks for pointing it out though, made me understand the promises better 

<img width="807" alt="Screen Shot 2019-05-04 at 12 28 27 PM" src="https://user-images.githubusercontent.com/13354855/57183317-2355f200-6e68-11e9-836b-eeddabd83980.png">

Reply:
Okay cool, just wanted to make sure because i know sequelize is all promise based so just wanted to make sure it wasnt going to the catch if it didnt find a user in the find one function!